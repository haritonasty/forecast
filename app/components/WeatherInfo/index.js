/**
 *
 * WeatherInfo
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { fontSize } from '../../utils/style';

import WindIcon from '../WindIcon';
import HumidityIcon from '../HumidityIcon';
import PrecipitationIcon from '../PrecipitationIcon';

const List = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  @media (max-width: 700px) {
    justify-content: space-around;
  }
`;

const Item = styled.li`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
`;

const Name = styled.span`
  font-size: ${fontSize.small}px;
`;

const Value = styled.span`
  font-size: ${fontSize.medium}px;
`;

const WrapperIcon = styled.div`
  width: 40px;
  height: 40px;
  margin: 20px 0;
`;

function WeatherInfo({ info }) {
  return (
    <List>
      <Item>
        <Name>Humidity</Name>
        <WrapperIcon>
          <HumidityIcon />
        </WrapperIcon>
        {info && <Value>{info.humidity}%</Value>}
      </Item>
      <Item>
        <Name>Wind</Name>
        <WrapperIcon>
          <WindIcon />
        </WrapperIcon>
        {info && (
          <Value>
            {info.wind}
            m/s
          </Value>
        )}
      </Item>
      <Item>
        <Name>Precipitation</Name>
        <WrapperIcon>
          <PrecipitationIcon />
        </WrapperIcon>
        {info && (
          <Value>
            {info.precipitation}
            mm
          </Value>
        )}
      </Item>
    </List>
  );
}

WeatherInfo.propTypes = {
  info: PropTypes.object,
};

export default WeatherInfo;
