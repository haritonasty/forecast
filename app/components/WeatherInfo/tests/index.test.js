import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import WeatherInfo from '../index';

describe('<WeatherInfo />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<WeatherInfo />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should match the snapshot with info', () => {
    const props = { info: { humidity: 5, wind: 40, precipitation: 0 } };
    const renderedComponent = renderer
      .create(<WeatherInfo {...props} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
