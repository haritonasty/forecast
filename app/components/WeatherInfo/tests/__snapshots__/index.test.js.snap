// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherInfo /> should match the snapshot 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c2 {
  font-size: 14px;
}

.c3 {
  width: 40px;
  height: 40px;
  margin: 20px 0;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <span
      className="c2"
    >
      Humidity
    </span>
    <div
      className="c3"
    >
      <svg
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M291 220l-20-39a9 9 0 1 0-15 8l19 38a114 114 0 0 1-67 147c-12 4-24 6-36 7A114 114 0 0 1 68 227L172 27l70 136a9 9 0 0 0 15-8L179 5a9 9 0 0 0-15 0L53 220h-1a131 131 0 0 0 77 170 131 131 0 0 0 43 8 131 131 0 0 0 119-178zM464 278l-1-1-19-38a9 9 0 0 0-15 8l19 38a52 52 0 0 1-47 69 52 52 0 0 1-47-69l47-91 15 30a9 9 0 0 0 16-8l-24-45a9 9 0 0 0-15 0l-55 106v1a69 69 0 0 0 40 89 69 69 0 0 0 23 5 69 69 0 0 0 63-94zM354 445l-37-71a9 9 0 0 0-15 0l-37 71v1a49 49 0 0 0 45 66 49 49 0 0 0 44-67zm-34 48l-10 2a32 32 0 0 1-29-43l29-55 28 55c6 17-2 35-18 41z"
        />
      </svg>
    </div>
  </li>
  <li
    className="c1"
  >
    <span
      className="c2"
    >
      Wind
    </span>
    <div
      className="c3"
    >
      <svg
        viewBox="0 -3 411 410"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M233 141h-4c-3 0-6-3-6-7 0-3 1-65 44-106a50 50 0 0 1 85 37c0 13-5 25-14 34-4 6-34 42-105 42zm44-104c-31 31-38 74-40 90 65-1 90-36 90-36l1-1a36 36 0 0 0 1-52 36 36 0 0 0-52-1zm0 0M7 342l-4-1c-3-3-4-7-2-10 1-1 81-109 205-89 122 19 164 18 185-30 12-25 6-46-15-57-20-10-51-5-64 19a7 7 0 0 1-12-6c17-32 56-38 82-25 21 10 39 36 22 74-27 59-82 57-200 38-116-18-191 83-192 84-1 2-3 3-5 3zm0 0"
        />
        <path
          d="M16 270a7 7 0 0 1-5-11c3-4 71-85 189-75a7 7 0 1 1-1 13c-110-9-177 69-177 70l-6 3zm0 0M284 406l-9-2c-16-3-27-14-33-33a7 7 0 0 1 12-4c5 14 13 22 24 24 8 2 17 0 24-5 7-6 12-15 12-25 0-15-9-29-22-36-15-9-44-19-93-22-94-4-108 57-108 58-1 3-5 6-8 5-4-1-6-4-6-8 0-1 17-73 123-68 41 2 75 9 99 23 17 9 28 28 28 48 0 14-6 27-17 35-7 6-16 10-26 10zm0 0M267 111l-3-1c-3-2-4-6-3-9 2-3 36-75 100-100a7 7 0 0 1 5 13c-59 22-92 92-93 93-1 2-3 4-6 4zm0 0"
        />
      </svg>
    </div>
  </li>
  <li
    className="c1"
  >
    <span
      className="c2"
    >
      Precipitation
    </span>
    <div
      className="c3"
    >
      <svg
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M503 141c-14 0-21-6-29-14-10-9-20-18-41-18s-32 9-42 18c-8 8-15 14-29 14s-21-6-29-14c-10-9-21-18-42-18s-31 9-41 18c-8 8-15 14-29 14-15 0-21-6-30-14-9-9-20-18-41-18s-32 9-41 18c-9 8-15 14-30 14s-21-6-29-14c-10-9-20-18-41-18a9 9 0 1 0 0 17c14 0 21 6 29 14 10 8 20 18 41 18s32-10 42-18c8-8 15-14 29-14s21 6 29 14c10 8 21 18 42 18s31-10 41-18c8-8 15-14 29-14s21 6 30 14 20 18 41 18 32-10 41-18 15-14 30-14c14 0 21 6 29 14 10 8 20 18 41 18a9 9 0 1 0 0-17zM503 223c-14 0-21-6-29-14-10-9-20-19-41-19s-32 10-42 19c-8 8-15 14-29 14s-21-6-29-14c-10-9-21-19-42-19s-31 10-41 19c-8 8-15 14-29 14-15 0-21-6-30-14l-3-3a9 9 0 0 0-12 13l3 3c10 8 21 18 42 18s31-10 41-18c8-8 15-14 29-14s21 6 30 14 20 18 41 18 32-10 41-18 15-14 30-14c14 0 21 6 29 14 10 8 20 18 41 18a9 9 0 1 0 0-17zM162 192l-12-2c-21 0-32 10-41 19-9 8-15 14-30 14s-21-6-29-14c-10-9-20-19-41-19a9 9 0 1 0 0 18c14 0 21 6 29 14 10 8 20 18 41 18s32-10 42-18c8-8 15-14 29-14l9 1a9 9 0 0 0 3-17zM503 386c-14 0-21-6-29-14-10-8-20-18-41-18s-32 10-42 18c-8 8-15 14-29 14s-21-6-29-14c-10-8-21-18-42-18s-31 10-41 18c-8 8-15 14-29 14-15 0-21-6-30-14s-20-18-41-18-32 10-41 18-15 14-30 14a9 9 0 1 0 0 17c21 0 32-9 42-18 8-8 15-14 29-14s21 6 29 14c10 9 21 18 42 18s31-9 41-18c8-8 15-14 29-14s21 6 30 14c9 9 20 18 41 18s32-9 41-18c9-8 15-14 30-14 14 0 21 6 29 14 10 9 20 18 41 18a9 9 0 1 0 0-17zM59 380l-9-8c-10-8-20-18-41-18a9 9 0 1 0 0 17c14 0 21 6 29 14l11 9a9 9 0 0 0 12-2c3-4 2-9-2-12zM503 304c-14 0-21-6-29-14-10-8-20-18-41-18s-32 10-42 18a9 9 0 1 0 12 13c9-8 15-14 30-14 14 0 21 6 29 14 10 9 20 19 41 19a9 9 0 1 0 0-18zM382 310c-1-5-6-8-10-7l-10 1c-14 0-21-6-29-14-10-8-21-18-42-18s-31 10-41 18c-8 8-15 14-29 14-15 0-21-6-30-14s-20-18-41-18-32 10-41 18-15 14-30 14-21-6-29-14c-10-8-20-18-41-18a9 9 0 1 0 0 17c14 0 21 6 29 14 10 9 20 19 41 19s32-10 42-19c8-8 15-14 29-14s21 6 29 14c10 9 21 19 42 19s31-10 41-19c8-8 15-14 29-14s21 6 30 14c9 9 20 19 41 19l14-2c5-1 7-6 6-10z"
        />
      </svg>
    </div>
  </li>
</ul>
`;

exports[`<WeatherInfo /> should match the snapshot with info 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c2 {
  font-size: 14px;
}

.c4 {
  font-size: 16px;
}

.c3 {
  width: 40px;
  height: 40px;
  margin: 20px 0;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <span
      className="c2"
    >
      Humidity
    </span>
    <div
      className="c3"
    >
      <svg
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M291 220l-20-39a9 9 0 1 0-15 8l19 38a114 114 0 0 1-67 147c-12 4-24 6-36 7A114 114 0 0 1 68 227L172 27l70 136a9 9 0 0 0 15-8L179 5a9 9 0 0 0-15 0L53 220h-1a131 131 0 0 0 77 170 131 131 0 0 0 43 8 131 131 0 0 0 119-178zM464 278l-1-1-19-38a9 9 0 0 0-15 8l19 38a52 52 0 0 1-47 69 52 52 0 0 1-47-69l47-91 15 30a9 9 0 0 0 16-8l-24-45a9 9 0 0 0-15 0l-55 106v1a69 69 0 0 0 40 89 69 69 0 0 0 23 5 69 69 0 0 0 63-94zM354 445l-37-71a9 9 0 0 0-15 0l-37 71v1a49 49 0 0 0 45 66 49 49 0 0 0 44-67zm-34 48l-10 2a32 32 0 0 1-29-43l29-55 28 55c6 17-2 35-18 41z"
        />
      </svg>
    </div>
    <span
      className="c4"
    >
      5
      %
    </span>
  </li>
  <li
    className="c1"
  >
    <span
      className="c2"
    >
      Wind
    </span>
    <div
      className="c3"
    >
      <svg
        viewBox="0 -3 411 410"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M233 141h-4c-3 0-6-3-6-7 0-3 1-65 44-106a50 50 0 0 1 85 37c0 13-5 25-14 34-4 6-34 42-105 42zm44-104c-31 31-38 74-40 90 65-1 90-36 90-36l1-1a36 36 0 0 0 1-52 36 36 0 0 0-52-1zm0 0M7 342l-4-1c-3-3-4-7-2-10 1-1 81-109 205-89 122 19 164 18 185-30 12-25 6-46-15-57-20-10-51-5-64 19a7 7 0 0 1-12-6c17-32 56-38 82-25 21 10 39 36 22 74-27 59-82 57-200 38-116-18-191 83-192 84-1 2-3 3-5 3zm0 0"
        />
        <path
          d="M16 270a7 7 0 0 1-5-11c3-4 71-85 189-75a7 7 0 1 1-1 13c-110-9-177 69-177 70l-6 3zm0 0M284 406l-9-2c-16-3-27-14-33-33a7 7 0 0 1 12-4c5 14 13 22 24 24 8 2 17 0 24-5 7-6 12-15 12-25 0-15-9-29-22-36-15-9-44-19-93-22-94-4-108 57-108 58-1 3-5 6-8 5-4-1-6-4-6-8 0-1 17-73 123-68 41 2 75 9 99 23 17 9 28 28 28 48 0 14-6 27-17 35-7 6-16 10-26 10zm0 0M267 111l-3-1c-3-2-4-6-3-9 2-3 36-75 100-100a7 7 0 0 1 5 13c-59 22-92 92-93 93-1 2-3 4-6 4zm0 0"
        />
      </svg>
    </div>
    <span
      className="c4"
    >
      40
      m/s
    </span>
  </li>
  <li
    className="c1"
  >
    <span
      className="c2"
    >
      Precipitation
    </span>
    <div
      className="c3"
    >
      <svg
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M503 141c-14 0-21-6-29-14-10-9-20-18-41-18s-32 9-42 18c-8 8-15 14-29 14s-21-6-29-14c-10-9-21-18-42-18s-31 9-41 18c-8 8-15 14-29 14-15 0-21-6-30-14-9-9-20-18-41-18s-32 9-41 18c-9 8-15 14-30 14s-21-6-29-14c-10-9-20-18-41-18a9 9 0 1 0 0 17c14 0 21 6 29 14 10 8 20 18 41 18s32-10 42-18c8-8 15-14 29-14s21 6 29 14c10 8 21 18 42 18s31-10 41-18c8-8 15-14 29-14s21 6 30 14 20 18 41 18 32-10 41-18 15-14 30-14c14 0 21 6 29 14 10 8 20 18 41 18a9 9 0 1 0 0-17zM503 223c-14 0-21-6-29-14-10-9-20-19-41-19s-32 10-42 19c-8 8-15 14-29 14s-21-6-29-14c-10-9-21-19-42-19s-31 10-41 19c-8 8-15 14-29 14-15 0-21-6-30-14l-3-3a9 9 0 0 0-12 13l3 3c10 8 21 18 42 18s31-10 41-18c8-8 15-14 29-14s21 6 30 14 20 18 41 18 32-10 41-18 15-14 30-14c14 0 21 6 29 14 10 8 20 18 41 18a9 9 0 1 0 0-17zM162 192l-12-2c-21 0-32 10-41 19-9 8-15 14-30 14s-21-6-29-14c-10-9-20-19-41-19a9 9 0 1 0 0 18c14 0 21 6 29 14 10 8 20 18 41 18s32-10 42-18c8-8 15-14 29-14l9 1a9 9 0 0 0 3-17zM503 386c-14 0-21-6-29-14-10-8-20-18-41-18s-32 10-42 18c-8 8-15 14-29 14s-21-6-29-14c-10-8-21-18-42-18s-31 10-41 18c-8 8-15 14-29 14-15 0-21-6-30-14s-20-18-41-18-32 10-41 18-15 14-30 14a9 9 0 1 0 0 17c21 0 32-9 42-18 8-8 15-14 29-14s21 6 29 14c10 9 21 18 42 18s31-9 41-18c8-8 15-14 29-14s21 6 30 14c9 9 20 18 41 18s32-9 41-18c9-8 15-14 30-14 14 0 21 6 29 14 10 9 20 18 41 18a9 9 0 1 0 0-17zM59 380l-9-8c-10-8-20-18-41-18a9 9 0 1 0 0 17c14 0 21 6 29 14l11 9a9 9 0 0 0 12-2c3-4 2-9-2-12zM503 304c-14 0-21-6-29-14-10-8-20-18-41-18s-32 10-42 18a9 9 0 1 0 12 13c9-8 15-14 30-14 14 0 21 6 29 14 10 9 20 19 41 19a9 9 0 1 0 0-18zM382 310c-1-5-6-8-10-7l-10 1c-14 0-21-6-29-14-10-8-21-18-42-18s-31 10-41 18c-8 8-15 14-29 14-15 0-21-6-30-14s-20-18-41-18-32 10-41 18-15 14-30 14-21-6-29-14c-10-8-20-18-41-18a9 9 0 1 0 0 17c14 0 21 6 29 14 10 9 20 19 41 19s32-10 42-19c8-8 15-14 29-14s21 6 29 14c10 9 21 19 42 19s31-10 41-19c8-8 15-14 29-14s21 6 30 14c9 9 20 19 41 19l14-2c5-1 7-6 6-10z"
        />
      </svg>
    </div>
    <span
      className="c4"
    >
      0
      mm
    </span>
  </li>
</ul>
`;
